canary/generator/transformators/transformers_categorical.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC
+from copy import deepcopy
+
+import numpy as np
+
+from .base import Transformer
+
+
+class TransformerCategorical(Transformer, ABC):
+    def __init__(self):
+        super(TransformerCategorical, self).__init__(['categorical', 'enumerated',
+                                                      'boolean', 'flag'])
+
+
+class AddToSomeBucketTransformer(TransformerCategorical):
+    def __init__(self, percentage=None, bucket_number=None, change_date=None):
+        """anomaly, works on buckets!!!"""
+        super(AddToSomeBucketTransformer, self).__init__()
+        self.is_bucket = False if bucket_number is None else True
+        self.is_percentage = False if percentage is None else True
+        self.is_date = False if change_date is None else True
+        self.percentage = percentage
+        self.bucket_number = bucket_number
+        self.change_date = change_date
+
+    def transform(self, bucket_dict, y):
+        bucket_dict_copy = deepcopy(bucket_dict)
+        y_copy = deepcopy(y)
+        if not self.is_date:
+            self.change_date = np.random.choice(list(bucket_dict_copy['data'].keys()))
+        hist = bucket_dict_copy['data'][self.change_date]
+        if not self.is_bucket:
+            self.bucket_number = np.random.choice(
+                list(range(len(hist))))
+        if not self.percentage:
+            self.percentage = np.random.uniform(0.2, 0.6)
```

Comment:
How were this limits picked? What's the rationale behind them? Can you please document it? Also, but that's a nit, a value between [0,1] is a ratio, rather than a percentage :) It could be useful to rename the variable accordingly.

Reply:
Hard question. Do you accept 'it looked nice with those limits' answer? I'm thinking what to do with all of those magic numbers...

Reply:
Ehm no. What does "looked nice" mean here? Does it mean that the anomaly is well spread across the whole time series of histograms? Try to explain that a bit better and it should be ok.